internal/world/azure.go

```diff
@@ -0,0 +1,205 @@
+	keyVaultUrl    string
+	subscriptionId string
+	clientId       string
+	clientSecret   string
+	apiVersion     string
+	token          string
+}
+
+func (w *World) Azure() *Azure {
+	if w.azure != nil {
+		return w.azure
+	}
+	azureSubscriptionId := w.checkAzureEnv(AzureSubscriptionId)
+	azureClientId       := w.checkAzureEnv(AzureClientId)
+	azureClientSecret   := w.checkAzureEnv(AzureClientSecret)
+	azureKeyVaultUrl    := w.checkAzureEnv(AzureKeyVaultUrl)
+	azureApiVersion     := w.checkAzureEnv(AzureApiVersion)
+
+	if azureApiVersion == "" {
+		azureApiVersion = "7.0"
+	}
+
+	w.azure = &Azure{
+		logger:         w.logger,
+		KeyMapping:     make(map[string]string),
+		subscriptionId: azureSubscriptionId,
+		clientId:       azureClientId,
+		clientSecret:   azureClientSecret,
+		keyVaultUrl:    azureKeyVaultUrl,
+		apiVersion:     azureApiVersion,
+	}
+	return w.azure
+}
+
+func (w *World) checkAzureEnv(env string) string {
+	value := os.Getenv(env)
+	if w.logger != nil && value == "" {
+		w.logger.Warnf("%v not set.", env)
+	}
+	return value
+}
+
+func (a *Azure) Secret(path string) (string, error) {
+	prefixPath := fmt.Sprintf("%s%s", a.Prefix, path)
+	mapped, ok := a.KeyMapping[prefixPath]
+	if !ok {
+		mapped = path
+	}
+	// azure keyvault only allows alphanumeric chars and dashes, for easier use and backwards compatibility we
+	// replace "/" with "--" and use the double dashes as a separator
+	mapped = strings.Replace(mapped, "/", "--", -1)
+	err := a.getBearerToken()
+	if err != nil {
+		return "", errors.Wrap(err, "could not get access token from https://login.microsoftonline.com/")
+	}
+	latestSecretVersion, err := a.getLatestSecretVersion(mapped)
+	if err != nil {
+		return "", errors.Wrapf(err, "could not get secrets version for %s", mapped)
+	}
+	secret, err := a.getSecret(mapped, latestSecretVersion)
+	if err != nil {
+		return "", errors.Wrapf(err, "could not get secrets for %s", mapped)
+	}
+	return secret, nil
+}
+
+func (a *Azure) getSecret(path string, secretVersion string) (string, error) {
+	body, err := a.doVaultRequest(fmt.Sprintf("/secrets/%s/%s", path, secretVersion))
+	if err != nil {
+		return "", err
+	}
+	var entry KeyVaultEntry
+	err = json.Unmarshal(body, &entry)
+	if err != nil {
+		return "", err
+	}
+	return entry.Value, nil
+}
+
+func (a *Azure) getLatestSecretVersion(path string) (string, error) {
+	secretPath := path
+	body, err := a.doVaultRequest(fmt.Sprintf("/secrets/%s/versions", secretPath))
+	if err != nil {
+		return "", err
+	}
+	var secretVersions SecretVersions
+	err = json.Unmarshal(body, &secretVersions)
+	if err != nil {
+		return "", err
+	}
+	latestVersion := secretVersions.Value[0]
+	for _, v := range secretVersions.Value {
+		if v.Attributes.Created > latestVersion.Attributes.Created {
+			latestVersion = v
+		}
+	}
+	// version value is returned as an URL so we split it and return the last part which contains the version string
+	split := strings.Split(latestVersion.ID, "/")
```

Comment:
Is this guaranteed to contain a `/`?

Reply:
yes it always returns the value is an URL something like this

`https://keyvault.net/bla/key`